client/src/query.rs

```diff
@@ -0,0 +1,302 @@
+    ) -> Result<(IterableQueryOutputBatchBox, Option<Self::Cursor>), Self::Error> {
+        let request_head = self.get_query_request_head();
+
+        let request = QueryRequest::StartIterable(query);
+
+        let response = request_head.assemble(request).build()?.send()?;
+        let response = decode_iterable_query_response(&response)?;
+
+        let (batch, cursor) = response.into_parts();
+
+        let cursor = cursor.map(|cursor| ClientQueryCursor {
+            request_head,
+            cursor,
+        });
+
+        Ok((batch, cursor))
+    }
+
+    fn continue_iterable_query(
+        cursor: Self::Cursor,
+    ) -> Result<(IterableQueryOutputBatchBox, Option<Self::Cursor>), Self::Error> {
+        let ClientQueryCursor {
+            request_head,
+            cursor,
+        } = cursor;
+
+        let request = QueryRequest::ContinueIterable(cursor);
+
+        let response = request_head.assemble(request).build()?.send()?;
+        let response = decode_iterable_query_response(&response)?;
+
+        let (batch, cursor) = response.into_parts();
+
+        let cursor = cursor.map(|cursor| ClientQueryCursor {
+            request_head,
+            cursor,
+        });
+
+        Ok((batch, cursor))
+    }
+}
+
+impl Client {
+    /// Get a [`ClientQueryRequestHead`] - an object that can be used to make queries independently of the client.
+    ///
+    /// You probably do not want to use it directly, but rather use [`Client::query`] or [`Client::iter_query`].
+    fn get_query_request_head(&self) -> ClientQueryRequestHead {
+        ClientQueryRequestHead {
+            torii_url: self.torii_url.clone(),
+            headers: self.headers.clone(),
+            account_id: self.account.clone(),
+            key_pair: self.key_pair.clone(),
+        }
+    }
+
+    /// Execute a singular query and return the result
+    pub fn query<Q>(&self, query: Q) -> Result<Q::Output, ClientQueryError>
+    where
+        Q: SingularQuery,
+        SingularQueryBox: From<Q>,
+        Q::Output: TryFrom<SingularQueryOutputBox>,
+        <Q::Output as TryFrom<SingularQueryOutputBox>>::Error: Debug,
+    {
+        let query = SingularQueryBox::from(query);
+
+        let result = self.execute_singular_query(query)?;
+
+        Ok(result
+            .try_into()
+            .expect("BUG: iroha returned unexpected type in singular query"))
+    }
+
+    /// Build an iterable query and return a builder object
+    pub fn iter_query<Q, P>(&self, query: Q) -> IterableQueryBuilder<Self, Q, P>
+    where
+        Q: IterableQuery,
+        Q::Item: HasPredicateBox<PredicateBoxType = P>,
+    {
+        IterableQueryBuilder::new(self, query)
+    }
+
+    /// Make a request to continue an iterable query with the provided raw [`ForwardCursor`]
+    ///
+    /// You probably do not want to use this function, but rather use the [`Self::iter_query`] method to make a query and iterate over its results.
+    pub fn raw_continue_iterable_query(
```

Comment:
```suggestion
    fn raw_continue_iterable_query(
```

is there a reason to have it public?

Reply:
Yes, it is used in `live_query_is_dropped_after_smart_contract_end` test. It purposefully breaks abstraction, and this method enables it.